Foreground permission required in manifest for Android 9 / API 28

<!--
Hello and thanks for contributing! To help us diagnose your problem quickly, please:

 - Include a minimal demonstration of the bug, including code, logs, and screenshots.
 - Ensure you can reproduce the bug using the latest release.
 - Only post to report a bug or request a feature; direct all other questions to: https://stackoverflow.com/questions/tagged/mapbox
-->

**Android API: 28 (Android 9)**
**Mapbox Navigation SDK version: 0.24.0**

### Steps to trigger behavior

 1. Starting navigation activity via the launcher

### Expected behavior
- No crash / error
### Actual behavior
- Crashes due to a missing permission. 


Looks like the migration guide for api 28 states that a foreground permission is necessary for foreground services now. 

https://developer.android.com/about/versions/pie/android-9.0-migration#tya

Its not considered a dangerous permission so adding the permission to the manifest fixed the issue, just wanted to give you guys the heads up in case you wanna add it to any getting started guides/sample app. 


stack trace:
```
    java.lang.SecurityException: Permission Denial: startForeground from pid=23263, uid=10259 requires android.permission.FOREGROUND_SERVICE
        at android.os.Parcel.createException(Parcel.java:1950)
        at android.os.Parcel.readException(Parcel.java:1918)
        at android.os.Parcel.readException(Parcel.java:1868)
        at android.app.IActivityManager$Stub$Proxy.setServiceForeground(IActivityManager.java:5198)
        at android.app.Service.startForeground(Service.java:695)
        at com.mapbox.services.android.navigation.v5.navigation.NavigationService.startForegroundNotification(NavigationService.java:130)
        at com.mapbox.services.android.navigation.v5.navigation.NavigationService.startNavigation(NavigationService.java:66)
        at com.mapbox.services.android.navigation.v5.navigation.MapboxNavigation.onServiceConnected(MapboxNavigation.java:727)
        at android.app.LoadedApk$ServiceDispatcher.doConnected(LoadedApk.java:1730)
        at android.app.LoadedApk$ServiceDispatcher$RunConnection.run(LoadedApk.java:1762)
        at android.os.Handler.handleCallback(Handler.java:873)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loop(Looper.java:193)
        at android.app.ActivityThread.main(ActivityThread.java:6680)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
     Caused by: android.os.RemoteException: Remote stack trace:
        at com.android.server.am.ActivityManagerService.enforcePermission(ActivityManagerService.java:9189)
        at com.android.server.am.ActiveServices.setServiceForegroundInnerLocked(ActiveServices.java:1189)
        at com.android.server.am.ActiveServices.setServiceForegroundLocked(ActiveServices.java:870)
        at com.android.server.am.ActivityManagerService.setServiceForeground(ActivityManagerService.java:20458)
        at android.app.IActivityManager$Stub.onTransact(IActivityManager.java:976)
```

--- a/gradle/dependencies.gradle
+++ b/gradle/dependencies.gradle
@@ -2,8 +2,8 @@ ext {
 
   androidVersions = [
       minSdkVersion    : 14,
-      targetSdkVersion : 27,
-      compileSdkVersion: 27,
+      targetSdkVersion : 28,
+      compileSdkVersion: 28,
       buildToolsVersion: '28.0.3'
   ]
 
@@ -28,7 +28,7 @@ ext {
       espressoVersion    : '3.0.2',
       spoonRunner        : '1.6.2',
       commonsIO          : '2.6',
-      robolectric        : '3.8',
+      robolectric        : '4.0.2',
       lifecycle          : '1.1.1',
       picasso            : '2.71828',
       gmsLocation        : '15.0.1',

--- a/libandroid-navigation/src/main/AndroidManifest.xml
+++ b/libandroid-navigation/src/main/AndroidManifest.xml
@@ -1,6 +1,8 @@
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
           package="com.mapbox.services.android.navigation">
 
+    <uses-permission android:name="android.permission.FOREGROUND_SERVICE"/>
+
     <application>
         <service android:name="com.mapbox.services.android.navigation.v5.navigation.NavigationService"/>
         <!-- Include the telemetry service to simplify set up (https://www.mapbox.com/telemetry) -->

--- a/libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/milestone/TriggerPropertyTest.java
+++ b/libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/milestone/TriggerPropertyTest.java
@@ -5,7 +5,6 @@ import com.google.gson.GsonBuilder;
 import com.mapbox.api.directions.v5.DirectionsAdapterFactory;
 import com.mapbox.api.directions.v5.models.DirectionsResponse;
 import com.mapbox.api.directions.v5.models.DirectionsRoute;
-import com.mapbox.services.android.navigation.BuildConfig;
 import com.mapbox.services.android.navigation.v5.BaseTest;
 import com.mapbox.services.android.navigation.v5.routeprogress.RouteProgress;
 
@@ -14,10 +13,8 @@ import junit.framework.Assert;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.robolectric.RobolectricTestRunner;
-import org.robolectric.annotation.Config;
 
 @RunWith(RobolectricTestRunner.class)
-@Config(constants = BuildConfig.class)
 public class TriggerPropertyTest extends BaseTest {
 
   private static final String ROUTE_FIXTURE = "directions_v5_precision_6.json";

--- a/libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/milestone/TriggerTest.java
+++ b/libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/milestone/TriggerTest.java
@@ -5,7 +5,6 @@ import com.google.gson.GsonBuilder;
 import com.mapbox.api.directions.v5.DirectionsAdapterFactory;
 import com.mapbox.api.directions.v5.models.DirectionsResponse;
 import com.mapbox.api.directions.v5.models.DirectionsRoute;
-import com.mapbox.services.android.navigation.BuildConfig;
 import com.mapbox.services.android.navigation.v5.BaseTest;
 import com.mapbox.services.android.navigation.v5.routeprogress.RouteProgress;
 
@@ -14,10 +13,8 @@ import junit.framework.Assert;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.robolectric.RobolectricTestRunner;
-import org.robolectric.annotation.Config;
 
 @RunWith(RobolectricTestRunner.class)
-@Config(constants = BuildConfig.class)
 public class TriggerTest extends BaseTest {
 
   private static final String ROUTE_FIXTURE = "directions_v5_precision_6.json";

--- a/libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/navigation/NavigationEventDispatcherTest.java
+++ b/libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/navigation/NavigationEventDispatcherTest.java
@@ -10,7 +10,6 @@ import com.mapbox.android.core.location.LocationEngine;
 import com.mapbox.api.directions.v5.DirectionsAdapterFactory;
 import com.mapbox.api.directions.v5.models.DirectionsResponse;
 import com.mapbox.api.directions.v5.models.DirectionsRoute;
-import com.mapbox.services.android.navigation.BuildConfig;
 import com.mapbox.services.android.navigation.v5.BaseTest;
 import com.mapbox.services.android.navigation.v5.milestone.BannerInstructionMilestone;
 import com.mapbox.services.android.navigation.v5.milestone.Milestone;
@@ -28,7 +27,6 @@ import org.junit.runner.RunWith;
 import org.mockito.Mock;
 import org.mockito.MockitoAnnotations;
 import org.robolectric.RobolectricTestRunner;
-import org.robolectric.annotation.Config;
 
 import static org.junit.Assert.assertNotNull;
 import static org.mockito.Mockito.mock;
@@ -37,7 +35,6 @@ import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.when;
 
 @RunWith(RobolectricTestRunner.class)
-@Config(constants = BuildConfig.class, manifest = Config.DEFAULT_MANIFEST_NAME)
 public class NavigationEventDispatcherTest extends BaseTest {
 
   private static final String PRECISION_6 = "directions_v5_precision_6.json";

--- a/libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/navigation/NavigationHelperTest.java
+++ b/libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/navigation/NavigationHelperTest.java
@@ -14,7 +14,6 @@ import com.mapbox.api.directions.v5.models.StepIntersection;
 import com.mapbox.core.constants.Constants;
 import com.mapbox.geojson.Point;
 import com.mapbox.geojson.utils.PolylineUtils;
-import com.mapbox.services.android.navigation.BuildConfig;
 import com.mapbox.services.android.navigation.v5.BaseTest;
 import com.mapbox.services.android.navigation.v5.routeprogress.CurrentLegAnnotation;
 import com.mapbox.services.android.navigation.v5.routeprogress.RouteLegProgress;
@@ -24,7 +23,6 @@ import com.mapbox.services.android.navigation.v5.routeprogress.RouteStepProgress
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.robolectric.RobolectricTestRunner;
-import org.robolectric.annotation.Config;
 
 import java.io.IOException;
 import java.util.ArrayList;
@@ -37,7 +35,6 @@ import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.when;
 
 @RunWith(RobolectricTestRunner.class)
-@Config(constants = BuildConfig.class, manifest = Config.DEFAULT_MANIFEST_NAME)
 public class NavigationHelperTest extends BaseTest {
 
   private static final String MULTI_LEG_ROUTE_FIXTURE = "directions_two_leg_route.json";

